engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/CompositeTableDataService.java

```diff
@@ -53,6 +53,37 @@ public CompositeTableDataService(@NotNull String name, @NotNull final ServiceSel
         this.serviceSelector = Require.neqNull(serviceSelector, "serviceSelector");
     }
 
+    @Override
+    @Nullable
+    public TableLocationProvider getRawTableLocationProvider(@NotNull TableKey tableKey,
+            @NotNull TableLocationKey tableLocationKey) {
+        final TableDataService[] services = serviceSelector.call(tableKey);
+
+        if (services == null || services.length == 0) {
+            return null;
+        }
+
+        TableLocationProvider tlp = null;
+        for (final TableDataService service : services) {
+            final TableLocationProvider tlpCandidate = service.getRawTableLocationProvider(tableKey, tableLocationKey);
+            if (tlpCandidate == null) {
+                return null;
+            }
```

Comment:
this looks wrong. You need to iterate all the services and find out which, if any, accept the keys.